docs/machine-learning/tutorials/taxi-fare.md

```diff
@@ -85,143 +85,143 @@ Remove the existing class definition and add the following code, which has two c
-```
+When the model is trained and evaluated, by default, the values in the **Label** column are considered as correct values to be predicted. As we want to predict the taxi trip fare, copy the `FareAmount` column into the **Label** column. To do that, use <xref:Microsoft.ML.Transforms.CopyColumnsEstimator> and add the following code:
 
-The last step in data preparation combines all of the feature columns into the **Features** column using the <xref:Microsoft.ML.Legacy.Transforms.ColumnConcatenator> transformation class. By default, a learning algorithm processes only features from the **Features** column. Add the following code:
+[!code-csharp[CopyColumnsEstimator](../../../samples/machine-learning/tutorials/TaxiFarePrediction/Program.cs#7 "Use the CopyColumnsEstimator")]
 
-```csharp
-pipeline.Add(new ColumnConcatenator("Features",
-                                    "VendorId",
-                                    "RateCode",
-                                    "PassengerCount",
-                                    "TripDistance",
-                                    "PaymentType"));
-```
+The algorithm that trains the model requires **numeric** features, so you have to transform the categorical data (`VendorId`, `RateCode`, and `PaymentType`) values into numbers. To do that, use <xref:Microsoft.ML.Transforms.Categorical.OneHotEncodingEstimator>, which assigns different numeric key values to the different values in each of the columns, and add the following code:
 
-Notice that the `TripTime` column, which corresponds to the `trip_time_in_secs` column in the data set file, isn't included. You already determined that it isn't a useful prediction feature.
+[!code-csharp[OneHotEncodingEstimator](../../../samples/machine-learning/tutorials/TaxiFarePrediction/Program.cs#8 "Use the OneHotEncodingEstimator")]
 
-> [!NOTE]
-> These steps must be added to the pipeline in the order specified above for successful execution.
+The last step in data preparation combines all of the feature columns into the **Features** column using the <xref:Microsoft.ML.Transforms.ColumnConcatenatingEstimator> transformation class. By default, a learning algorithm processes only features from the **Features** column. Add the following code:
 
-## Choose a learning algorithm
+[!code-csharp[ColumnConcatenatingEstimator](../../../samples/machine-learning/tutorials/TaxiFarePrediction/Program.cs#9 "Use the ColumnConcatenatingEstimator")]
 
-After adding the data to the pipeline and transforming it into the correct input format, you select a learning algorithm (**learner**). The learner trains the model. You chose a **regression** task for this problem, so you use a <xref:Microsoft.ML.Legacy.Trainers.FastTreeRegressor> learner, which is one of the regression learners provided by ML.NET.
+Notice that the `TripTime` column, which corresponds to the `trip_time_in_secs` column in the data set file, isn't included. We already determined that it isn't a useful prediction feature.
```

Comment:
>We already determined that it isn't a useful prediction feature. [](start = 125, length = 64)

I was able to use all the features with good results. Please try the following pipeline and compare the metrics. It should be much better. Of course, please change all the "new ..." to mlContext equivalent. There are examples in the cookbook.

var pipeline = new CopyColumnsEstimator(mlcontext, "FareAmount", "Label")
                                    .Append(new CategoricalEstimator(mlcontext, "VendorId"))
                                    .Append(new CategoricalEstimator(mlcontext, "RateCode"))
                                    .Append(new CategoricalEstimator(mlcontext, "PaymentType"))
                                    .Append(new Normalizer(mlcontext, "PassengerCount", Normalizer.NormalizerMode.MeanVariance))
                                    .Append(new Normalizer(mlcontext, "TripTime", Normalizer.NormalizerMode.MeanVariance))
                                    .Append(new Normalizer(mlcontext, "TripDistance", Normalizer.NormalizerMode.MeanVariance))
                                    .Append(new ConcatEstimator(mlcontext, "Features", "VendorId", "RateCode", "PassengerCount", "TripTime", "TripDistance", "PaymentType"));

Reply:
Will do. The cookbook was not updated when this was updated this weekend.